### PR TITLE
Remove type parameter from CIR.mapping(type, object...)

### PR DESCRIPTION
**Pull request by romseygeek**
This commit removes the `type` parameter from `CreateIndexRequest.mapping(type, object...)`,
and the associated delegating method on `CreateIndexRequestBuilder`.   To make migration
simpler, the method on `CreateIndexRequest` is renamed to `mappingFromSimplifiedDef`, and
on `CreateIndexRequestBuilder` to `setMapping`; this should help the compiler catch all 
necessary changes on upgrades.

Relates to #41059

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/action/admin/indices/create/CreateIndexRequest.java [hunk 1]
```diff
@@ -263,6 +249,21 @@ public CreateIndexRequest mapping(Map<String, ?> source) {
         return mapping(MapperService.SINGLE_MAPPING_NAME, source);
     }
 
+    /**
+     * A specialized simplified mapping source method, takes the form of simple properties definition:
+     * ("field1", "type=string,store=true").
+     */
+    public CreateIndexRequest mappingFromSimplifiedDef(Object... source) {
```
Comment (jtibshirani) [inline server/src/main/java/org/elasticsearch/action/admin/indices/create/CreateIndexRequest.java]: A suggestion -- naming it `simpleMapping` could be more concise/ readable.

**Review by jtibshirani — APPROVED** [review]
This looks good to me, just left a suggestion around naming. It's great that we made sure to change the method names, so callers will see compile errors instead of confusing runtime errors.

**Comment by romseygeek** [comment]
@elasticmachine run elasticsearch-ci/2

File: server/src/main/java/org/elasticsearch/action/admin/indices/create/CreateIndexRequest.java (hunk 1 above)
Comment (romseygeek) [reply server/src/main/java/org/elasticsearch/action/admin/indices/create/CreateIndexRequest.java]: ++, good idea

**Pull request merged**